src/components/PlayingField/index.tsx

```diff
@@ -1,53 +1,174 @@
+}
+
+interface PokemonType {
+  name: string;
+  id: number;
+  sprites: {
+    front_default: string;
+  };
+}
 
 function PlayingField() {
-  const [imageHistory, setImageHistory] = useState<string>("");
-  const [points, setPoints] = useState<number>(0);
-
-  let cardList = [
-    blastoise,
-    bulbassaur,
-    caterpie,
-    charizard,
-    charmander,
-    charmeleon,
-    blastoise,
-    bulbassaur,
-    caterpie,
-    charizard,
-    charmander,
-    charmeleon,
-  ];
-
-  const handleClick = (image: string) => {
-    if (image === imageHistory) {
-      setPoints(points + 1);
+  const [isWaitingTimeout, setIsWaitingTimeout] = useState<boolean>(false);
+  const [lastCard, setLastCard] = useState<Card | null>(null);
+  const [pointsState, setPointsState] = useState<number>(0);
+  const [cardList, setCardList] = useState<Card[]>([]);
+  const [pokemon, setPokemon] = useState<PokemonType[]>([]);
+
+  function handleClick(clickedCard: Card) {
+    if (
+      cardList.find((card) => card.id === clickedCard.id)?.pairs ||
+      isWaitingTimeout
+    ) {
+      return;
+    }
+
+    let newCardList = [...cardList];
+
+    newCardList = newCardList.map((card) => {
+      if (card.id === clickedCard.id) {
+        return { ...card, isFlipped: !card.isFlipped };
+      } else {
+        return card;
+      }
+    });
+
+    if (clickedCard.img === lastCard?.img && clickedCard.id !== lastCard?.id) {
+      setPointsState(pointsState + 1);
+      setLastCard(null);
+      newCardList = newCardList.map((card) => {
+        if (card.img === clickedCard.img) {
+          return { ...card, pairs: true };
+        } else {
+          return card;
+        }
+      });
     } else {
-      setImageHistory(image);
+      if (lastCard) {
+        setIsWaitingTimeout(true);
+        setTimeout(() => {
+          newCardList = newCardList.map((card) => {
+            if (card.id === clickedCard.id || card.id === lastCard.id) {
+              return { ...card, isFlipped: false };
+            } else {
+              return card;
+            }
+          });
+
+          setCardList(newCardList);
+          setLastCard(null);
+          setIsWaitingTimeout(false);
+        }, 1000);
+      }
+      setLastCard(clickedCard);
     }
-  };
+
+    setCardList(newCardList);
+  }
+
+  function shuffleArray(array: Card[]) {
+    let currentIndex = array.length; //12
+    let randomIndex;
+
+    while (currentIndex !== 0) {
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex--;
+
+      [array[currentIndex], array[randomIndex]] = [
+        array[randomIndex],
+        array[currentIndex],
+      ];
+    }
+
+    return array;
+  }
+
+  function transformPokemonToCards(results: PokemonType[]): Card[] {
+    return results.map((pokemon) => ({
+      id: pokemon.id,
+      img: pokemon.sprites.front_default,
+      isFlipped: false,
+      pairs: false,
+    }));
+  }
+
+  async function fetchPokemons(pokemonNames: string[]) {
```

Comment:
Convert this to a `useCallback` and then add this function as a dependency of the `useEffect`